Compilation with recent gcc flags a null pointer dereference warning
which is promoted to an error and causes failure to build.

Fixed in upstream patch by Mark Wielaard:

https://lists.yoctoproject.org/g/poky/message/12815

diff --git a/src/elflint.c b/src/elflint.c
index 6a94683..dae360b 100644
--- a/src/elflint.c
+++ b/src/elflint.c
@@ -4100,7 +4100,7 @@ section [%2zu] '%s' has type NOBITS but is read from the file in segment of prog
 			    bad = (databits == NULL
 				   || databits->d_size != shdr->sh_size);
 			    for (size_t idx = 0;
-				 idx < databits->d_size && ! bad;
+				 ! bad && idx < databits->d_size;
 				 idx++)
 			      bad = ((char *) databits->d_buf)[idx] != 0;
 
